app/views/job_invocations/_tab_hosts.html.erb

```diff
@@ -12,7 +12,7 @@
     <tbody>
       <% job_invocation.targeting.hosts.each do |host| %>
         <tr>
-          <% task = job_invocation.sub_tasks.select { |task| task.locks.where(:resource_type => 'Host::Managed', :resource_id => host.id).any? }.first %>
+          <% task = job_invocation.sub_tasks.joins(:locks).where("#{ForemanTasks::Lock.table_name}.resource_type" => 'Host::Managed', "#{ForemanTasks::Lock.table_name}.resource_id" => host.id).first %>
```

Comment:
you can probably fetch all host_ids instead of per host, e.g.:
```ruby
job.sub_tasks.joins(:locks).where(
   'foreman_tasks_locks.resource_type' => 'Host::Managed',    
   'foreman_tasks_locks.resource_id'   => job.targeting.hosts)
```